### ISSUE TITLE
Connection reset by peer and grin server will not connect to peers or download headers

### ISSUE THREAD

**Issue by BLOCKCHAINSMOKER** [mimblewimble/grin#3203]
**Describe the bug**
I have tried a snap install grin and a github install grin and with both installs I cannot connect to any peers or get any headers. 

**To Reproduce**
1. Download grin through snap or through github
2. reboot
3. run: grin
4. "Waiting for peers" connected 0 disk usage 0

```
20200121 15:49:08.155 WARN grin_servers::grin::server - Grin server started.
20200121 15:49:08.156 INFO grin_servers::common::types - DandelionEpoch: next_epoch: is_stem: true (90%), relay: None
20200121 15:49:08.453 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.713.mw
20200121 15:49:08.470 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.grin.lesceller.com
20200121 15:49:08.694 DEBUG grin_servers::grin::seed - Failed to resolve seed "mainnet.seed.grin.lesceller.com" got error Custom { kind: Other, error: "failed to lookup address information: Temporary failure in name resolution" }
20200121 15:49:08.694 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.grin.prokapi.com
20200121 15:49:09.094 DEBUG grin_servers::grin::seed - Failed to resolve seed "mainnet.seed.grin.prokapi.com" got error Custom { kind: Other, error: "failed to lookup address information: No address associated with hostname" }
20200121 15:49:09.094 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns grinseed.yeastplume.org
20200121 15:49:09.342 DEBUG grin_servers::grin::seed - Failed to resolve seed "grinseed.yeastplume.org" got error Custom { kind: Other, error: "failed to lookup address information: Temporary failure in name resolution" }
20200121 15:49:09.342 DEBUG grin_servers::grin::seed - Retrieved seed addresses: [PeerAddr(V4(47.88.11.190:3414)), PeerAddr(V4(67.207.70.122:3414)), PeerAddr(V4(46.4.91.48:3414)), PeerAddr(V4(157.230.142.11:3414)), PeerAddr(V4(94.130.229.193:3414)), PeerAddr(V4(37.187.124.202:3414)), PeerAddr(V4(159.65.11.32:3414))]
20200121 15:49:09.342 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 15:49:09.576 INFO grin_store::lmdb - Resized database from 1048576 to 134217728
20200121 15:49:09.639 DEBUG grin_p2p::peer - connect: handshaking with Ok(V4(47.88.11.190:3414))
20200121 15:49:11.343 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 15:49:15.343 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 15:49:17.038 DEBUG grin_p2p::peer - connect: handshaking with Err(Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" }) failed with error: Connection(Os { code: 104, kind: ConnectionReset, message: "Connection reset by peer" })
20200121 15:49:17.038 DEBUG grin_p2p::peer - Error shutting down conn: Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" }
20200121 15:49:23.345 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 15:49:39.347 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 15:49:59.349 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 15:50:19.352 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 15:50:39.354 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
```

This last log is actually with a connection through a VPN. I recieved the same error logs without the VPN connections but I will produce them below. 

```
20200121 13:23:22.970 WARN grin_servers::grin::server - Grin server started.
20200121 13:23:22.970 INFO grin_servers::common::types - DandelionEpoch: next_epoch: is_stem: true (90%), relay: None
20200121 13:23:22.970 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.713.mw
20200121 13:23:22.985 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.grin.lesceller.com
20200121 13:23:23.000 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.grin.prokapi.com
20200121 13:23:23.000 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns grinseed.yeastplume.org
20200121 13:23:23.000 DEBUG grin_servers::grin::seed - Retrieved seed addresses: [PeerAddr(V4(47.88.11.190:3414)), PeerAddr(V4(46.4.91.48:3414)), PeerAddr(V4(37.187.124.202:3414)), PeerAddr(V4(157.230.142.11:3414)), PeerAddr(V4(94.130.229.193:3414)), PeerAddr(V4(67.207.70.122:3414)), PeerAddr(V4(159.65.11.32:3414)), PeerAddr(V4(5.135.184.54:3414)), PeerAddr(V4(35.223.12.108:3414)), PeerAddr(V4(35.181.35.93:3414)), PeerAddr(V4(145.239.4.27:3414)), PeerAddr(V4(95.216.163.175:3414)), PeerAddr(V4(35.211.201.137:3414)), PeerAddr(V4(157.230.48.217:3414)), PeerAddr(V4(18.204.166.78:3414)), PeerAddr(V4(5.9.152.75:3414)), PeerAddr(V4(34.207.234.131:3414)), PeerAddr(V4(159.69.37.136:3414)), PeerAddr(V4(109.74.202.16:3414))]
20200121 13:23:23.001 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 13:23:23.100 INFO grin_store::lmdb - Resized database from 1048576 to 134217728
20200121 13:23:23.110 DEBUG grin_p2p::peer - connect: handshaking with Ok(V4(35.181.35.93:3414))
20200121 13:23:23.199 DEBUG grin_p2p::peer - connect: handshaking with Ok(V4(18.204.166.78:3414))
20200121 13:23:23.317 DEBUG grin_p2p::peer - connect: handshaking with Ok(V4(47.88.11.190:3414))
20200121 13:23:25.001 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 13:23:29.002 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 13:23:31.326 DEBUG grin_p2p::peer - connect: handshaking with Err(Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" }) failed with error: Connection(Os { code: 104, kind: ConnectionReset, message: "Connection reset by peer" })
20200121 13:23:31.326 DEBUG grin_p2p::peer - Error shutting down conn: Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" }
20200121 13:23:37.003 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 13:23:53.005 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 13:24:13.007 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 13:24:33.010 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 13:24:53.012 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 13:25:13.015 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20200121 13:25:20.201 WARN grin::tui::ui - Shutdown in progress, please wait
20200121 13:25:20.201 DEBUG grin_chain::types - sync_state: sync_status: AwaitingPeers(true) -> Shutdown
20200121 13:25:21.016 INFO grin_servers::grin::server - connect_and_monitor thread stopped
20200121 13:25:21.016 INFO grin_servers::grin::server - sync thread stopped
20200121 13:25:21.016 INFO grin_servers::grin::server - dandelion_monitor thread stopped
20200121 13:25:21.016 WARN grin_servers::grin::server - Shutdown complete
20200121 15:49:08.123 INFO grin_util::logger - log4rs is initialized, file level: Debug, stdout level: Debug, min. level: Debug
20200121 15:49:08.123 INFO grin - Using configuration file at /home/grin/snap/grin/24/.grin/main/grin-server.toml
20200121 15:49:08.123 INFO grin - This is Grin version 3.0.0 (git v3.0.0), built for x86_64-unknown-linux-gnu by rustc 1.40.0 (73528e339 2019-12-16).
20200121 15:49:08.123 DEBUG grin - Built with profile "release", features "".
20200121 15:49:08.123 WARN grin::cmd::server - Starting GRIN in UI mode...
20200121 15:49:08.123 INFO grin_servers::grin::server - Starting server, genesis block: 40adad0aec27
20200121 15:49:08.123 DEBUG grin_store::lmdb - DB Mapsize for /home/grin/snap/grin/24/.grin/main/chain_data/lmdb is 134217728
20200121 15:49:08.123 DEBUG grin_store::leaf_set - bitmap 1 pos (18 bytes)
20200121 15:49:08.123 DEBUG grin_store::leaf_set - bitmap 1 pos (18 bytes)
20200121 15:49:08.123 DEBUG grin_chain::txhashset::bitmap_accumulator - applied 1 chunks from idx 0 to idx 0 (0ms)
20200121 15:49:08.134 DEBUG grin_chain::txhashset::txhashset - attempting to open kernel PMMR using ProtocolVersion(2) - SUCCESS
20200121 15:49:08.134 DEBUG grin_chain::txhashset::txhashset - Rewind header extension to 40adad0aec27 at 0 from 40adad0aec27 at 0
20200121 15:49:08.134 DEBUG grin_chain::txhashset::txhashset - Rewind extension to 40adad0aec27 at 0 from 40adad0aec27 at 0
20200121 15:49:08.134 DEBUG grin_chain::txhashset::bitmap_accumulator - applied 1 chunks from idx 0 to idx 0 (0ms)
20200121 15:49:08.134 DEBUG grin_chain::chain - init: rewinding and validating before we start... 40adad0aec27 at 0
20200121 15:49:08.153 DEBUG grin_chain::chain - rebuild_height_for_pos: nothing to be rebuilt
20200121 15:49:08.153 DEBUG grin_chain::chain - head: 17179869184 @ 0 [40adad0aec27]
20200121 15:49:08.153 DEBUG grin_chain::chain - header_head: 17179869184 @ 0 [40adad0aec27]
20200121 15:49:08.153 DEBUG grin_chain::chain - sync_head: 17179869184 @ 0 [40adad0aec27]
20200121 15:49:08.154 DEBUG grin_store::lmdb - DB Mapsize for /home/grin/snap/grin/24/.grin/main/chain_data/peer is 1048576
20200121 15:49:08.154 DEBUG grin_chain::types - sync_state: sync_status: Initial -> AwaitingPeers(true)
20200121 15:49:08.154 INFO grin_servers::grin::server - Starting rest apis at: 127.0.0.1:3413
20200121 15:49:08.154 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.grin.icu
20200121 15:49:08.154 WARN grin_api::handlers - Starting HTTP Node APIs server at 127.0.0.1:3413.
20200121 15:49:08.155 WARN grin_api::handlers - HTTP Node listener started.
20200121 15:49:08.155 INFO grin_servers::grin::server - Starting dandelion monitor: 127.0.0.1:3413
20200121 15:49:08.155 DEBUG grin_servers::grin::dandelion_monitor - Started Dandelion transaction monitor.
```

**Comment by BLOCKCHAINSMOKER** [comment]
Update! VPN connection worked after I deleted chain data! However the underlying issue remains that I was banned by the grin network. I think it is an IP ban. This must have been included in the new update. But why the censorship from the grin network? Any ideas

**Comment by antiochp** [comment]
Is it possible you were running a pre-HF node? One reason for a ban is sending "bad" block data and may be explained if you were to send a now invalid block post-HF.

Alternatively you may have been banned by not sending block headers when requested - but that would be unlikely to explain why you got banned by all your peers.

**Comment by BLOCKCHAINSMOKER** [comment]
@antiochp No I was running grin 3.0. At the moment I can only connect with a vpn. I think it is because the grin nodes are censoring me based on my current IP address which is Iran. What should I do to fix this.... run my own seed node?

**Comment by quentinlesceller** [comment]
Closing due to lack of activity. Feel free to open another issue if this is still relevant.
